### PR TITLE
update go to go1.20.6

### PR DESCRIPTION
**Pull request by thaJeztah**
- [x] depends on https://github.com/docker/cli/pull/4422

go1.20.6 (released 2023-07-11) includes a security fix to the net/http package, as well as bug fixes to the compiler, cgo, the cover tool, the go command, the runtime, and the crypto/ecdsa, go/build, go/printer, net/mail, and text/template packages. See the Go 1.20.6 milestone on our issue tracker for details.

https://github.com/golang/go/issues?q=milestone%3AGo1.20.6+label%3ACherryPickApproved

Full diff: https://github.com/golang/go/compare/go1.20.5...go1.20.6

These minor releases include 1 security fixes following the security policy:

net/http: insufficient sanitization of Host header

The HTTP/1 client did not fully validate the contents of the Host header. A maliciously crafted Host header could inject additional headers or entire requests. The HTTP/1 client now refuses to send requests containing an invalid Request.Host or Request.URL.Host value.

Thanks to Bartek Nowotarski for reporting this issue.

Includes security fixes for [CVE-2023-29406 ][1] and Go issue https://go.dev/issue/60374

[1]: https://github.com/advisories/GHSA-f8f7-69v5-w4vx

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by thaJeztah** [comment]
Rebased and moved out of draft, because https://github.com/docker/cli/pull/4422 was merged.

File: docker-bake.hcl [hunk 1]
```diff
@@ -1,5 +1,5 @@
 variable "GO_VERSION" {
-    default = "1.20.5"
+    default = "1.20.6"
```
Comment (crazy-max) [inline docker-bake.hcl:2]: We could use `null` value here to default with the one from Dockerfile: https://docs.docker.com/build/bake/reference/#targetargs

**Comment by thaJeztah** [comment]
Let me bring this one in

File: docker-bake.hcl (hunk 1 above)
Comment (thaJeztah) [reply docker-bake.hcl:2]: oh dang; missed your comment; is that supported everywhere now? Want to open a PR?

**Pull request merged**